### PR TITLE
WRN-1422: Update samples lint and build checking

### PR DESCRIPTION
**Pull request by YunhoLee0**
Enact-DCO-1.0-Signed-off-by: Yunho Lee (yunho2.lee@lge.com)

### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
Currently, the scripts for checking lint and build are targeting tutorials instead of enact-all-samples.
And recently samples for each theme were moved to this repository, but the scripts don't work with them.
It should be updated.

### Resolution
Rename enact-all-samples to all-samples.
Move and copy it under each theme folder.
Change all-samples app to work for its samples.
Update scripts for checking lint and build.
For sample apps using redux, make main.js separated from index.js so all-samples could import its App component.

### Links
WRN-1422

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,29 +3,39 @@ node_js:
...
+    - npm install
+    - npm run pack
+    - cd ../../
+    - echo -e "\x1b\x5b35;1m*** ui/all-samples build complete\x1b\x5b0m"
+    - echo -e "\x1b\x5b35;1m*** Starting my-theme-app build...\x1b\x5b0m"
+    - cd my-theme-app
+    - npm install
+    - npm run pack
+    - cd ../../
+    - echo -e "\x1b\x5b35;1m*** my-theme-app build complete\x1b\x5b0m"
```
Comment (MikyungKim) [inline .travis.yml:41]: Please add one empty line.

File: agate/pattern-virtuallist-preserving-focus/src/index.js [hunk 2]
```diff
@@ -1,17 +1,9 @@
 import {render} from 'react-dom';
-import {Provider} from 'react-redux';
 import 'web-animations-js';
 
-import App from './App';
-import configureStore from './store';
+import App from './main';
```
Comment (MikyungKim) [inline agate/pattern-virtuallist-preserving-focus/src/index.js:4]: Why this change needed?

File: moonstone/all-samples/src/components/ButtonToSamples/ButtonToSamples.module.less [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+			transform: translate(0px, 0px);
+		}
+	}
+}
+
+.buttonContainer {
+	display: flex;
+	flex-direction: row;
+	justify-content: flex-end;
+}
```
Comment (MikyungKim) [inline moonstone/all-samples/src/components/ButtonToSamples/ButtonToSamples.module.less:22]: Please add one empty line.

File: sandstone/all-samples/src/components/ButtonToSamples/ButtonToSamples.module.less [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+			transform: translate(0px, 0px);
+		}
+	}
+}
+
+.buttonContainer {
+	display: flex;
+	flex-direction: row;
+	justify-content: flex-end;
+}
```
Comment (MikyungKim) [inline sandstone/all-samples/src/components/ButtonToSamples/ButtonToSamples.module.less:22]: Please add one empty line.

File: scripts/build.js [hunk 5]
```diff
@@ -23,12 +23,11 @@ findApps()
 			if (file.parentDir) { // Ignore our own package.json
 				console.log(`Building ${file.parentDir}`);
 				shell.cd(file.fullParentDir);
-				shell.exec('npm install && npm run pack', {silent: true}, (code, stdout) => {
-					if (code) {
-						console.log(`${file.parentDir} failed to build: ${stdout}`);
-						error = true;
-					}
-				});
+				const result = shell.exec('npm install && npm run pack', {silent: true});
```
Comment (MikyungKim) [inline scripts/build.js:26]: Why this change needed?

File: ui/all-samples/src/components/ButtonToSamples/ButtonToSamples.module.less [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+			transform: translate(0px, 0px);
+		}
+	}
+}
+
+.buttonContainer {
+	display: flex;
+	flex-direction: row;
+	justify-content: flex-end;
+}
```
Comment (MikyungKim) [inline ui/all-samples/src/components/ButtonToSamples/ButtonToSamples.module.less:22]: Please add one empty line.

File: .travis.yml (hunk 1 above)
Comment (YunhoLee0) [reply .travis.yml:41]: OK

File: agate/pattern-virtuallist-preserving-focus/src/index.js (hunk 2 above)
Comment (YunhoLee0) [reply agate/pattern-virtuallist-preserving-focus/src/index.js:4]: The all-samples import this app. However, index.js can't be imported because it is a starting point where the app is rendered. So the App component needs to be imported, but the redux configuration is required for this sample. This behavior is performed on main.js, which is why it is separated from index.js.
This is the same pattern as other samples using redux.

File: moonstone/all-samples/src/components/ButtonToSamples/ButtonToSamples.module.less (hunk 3 above)
Comment (YunhoLee0) [reply moonstone/all-samples/src/components/ButtonToSamples/ButtonToSamples.module.less:22]: OK

File: sandstone/all-samples/src/components/ButtonToSamples/ButtonToSamples.module.less (hunk 4 above)
Comment (YunhoLee0) [reply sandstone/all-samples/src/components/ButtonToSamples/ButtonToSamples.module.less:22]: OK

File: scripts/build.js (hunk 5 above)
Comment (YunhoLee0) [reply scripts/build.js:26]: This change makes `shell.exec` work synchronously. By Travis testing, if the `shell.exec` work asynchronously, time out occurred. It seems the Travis is timed out when there is no output over 10 minutes. This was not an issue before cause the number of apps was less than now.

File: ui/all-samples/src/components/ButtonToSamples/ButtonToSamples.module.less (hunk 6 above)
Comment (YunhoLee0) [reply ui/all-samples/src/components/ButtonToSamples/ButtonToSamples.module.less:22]: OK

File: .travis.yml [hunk 7]
```diff
@@ -3,29 +3,40 @@ node_js:
...
+
```
Comment (MikyungKim) [inline .travis.yml:42]: 🤔 It's not resolved. Please remove the last line.

File: agate/all-samples/src/components/ButtonToSamples/ButtonToSamples.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
+import Button from '@enact/agate/Button';
+import {Link} from 'react-router-dom';
+
+import css from './ButtonToSamples.module.less';
+
+const ButtonToSamples = () => (
```
Comment (MikyungKim) [inline agate/all-samples/src/components/ButtonToSamples/ButtonToSamples.js:6]: Because of this component, every all-samples has 5way navigation issue.
Do you know what is this component for?
If you run the all-samples, we cannot navigate with 5way after entering a menu.

Comment (YunhoLee0) [reply agate/all-samples/src/components/ButtonToSamples/ButtonToSamples.js:6]: The enact-all-samples that was based on the moonstone had the same 5-way navigation issue.
I cannot find the reason now so created bug tickets. (WRN-3524, WRN-3523)

File: .travis.yml (hunk 7 above)
Comment (YunhoLee0) [reply .travis.yml]: It's fixed. :)

**Review by MikyungKim — APPROVED** [review]
LGTM

File: .travis.yml [hunk 9]
```diff
@@ -3,29 +3,39 @@ node_js:
...
-    # Temporarily check only two folders. (PLAT-128528)
     - echo -e "\x1b\x5b35;1m*** Starting eslint...\x1b\x5b0m"
     - npm run lint
     - echo -e "\x1b\x5b35;1m*** eslint complete\x1b\x5b0m"
-    - echo -e "\x1b\x5b35;1m*** Starting tutorial-hello-enact build...\x1b\x5b0m"
-    - cd tutorial-hello-enact
+    - echo -e "\x1b\x5b35;1m*** Starting agate/all-samples build...\x1b\x5b0m"
+    - cd agate/all-samples
     - npm install
     - npm run pack
```
Comment (jeonghee27) [inline .travis.yml:15]: Is it ok to skip checking for tutorial-hello-enact ?

File: agate/pattern-virtuallist-preserving-focus/src/index.js (hunk 2 above)
Comment (jeonghee27) [reply agate/pattern-virtuallist-preserving-focus/src/index.js:4]: Could you leave this on the Resolution of PR comment ?

Comment (YunhoLee0) [reply agate/pattern-virtuallist-preserving-focus/src/index.js:4]: I've added it to this PR's resolution section.

File: .travis.yml (hunk 9 above)
Comment (YunhoLee0) [reply .travis.yml:15]: Those apps are based on sandstone. So they are included and checked by sandstone/all-samples.

**Review by jeonghee27 — APPROVED** [review]
Good job
